framework/src/meshdivisions/CylindricalGridDivision.C

```diff
@@ -36,10 +36,8 @@ CylindricalGridDivision::validParams()
   params.addRangeCheckedParam<Real>(
       "r_min", 0, "r_min>=0", "Minimum radial coordinate (for a hollow cylinder)");
   params.addRequiredRangeCheckedParam<Real>("r_max", "r_max>0", "Maximum radial coordinate");
-  params.addRangeCheckedParam<Real>("cylinder_axial_min",
-                                    std::numeric_limits<Real>::min(),
-                                    "cylinder_axial_min>0",
-                                    "Minimum axial coordinate");
+  params.addRangeCheckedParam<Real>(
+      "cylinder_axial_min", 0, "cylinder_axial_min>=0", "Minimum axial coordinate");
```

Comment:
no that wont work.

The cylinder was previously infinite in the minus-axial direction. Now it s not only finite but of 0 height in the minus-axial direction

Reply:
we do need a big negative number here. It does not have to be -10^300 though. ANything pretty pretty big is good enough

Reply:
oh and the range check was wrong ... that was my fault!

Reply:
std::numeric_limits<Real>::lowest() would have been right then

I m afraid I never used min() before amd I guess I got it wrong

Reply:
-10^30 is probably big (small) enough tbh

Reply:
@GiudGiud -

Okay great, `std::numeric_limits<Real>::lowest()` actually works fine for `ss >> double` on ARM Mac.

I'll simply change it to that, and also fix the `_max_z == std::numeric_limits<Real>::max()` range check.

The default will now be `-1.79769e+308` (super negative) instead of `2.22507e-308` (positive almost zero)

The only issue now is the current range check function `cylinder_axial_min>0` in:

```
params.addRangeCheckedParam<Real>("cylinder_axial_min",
                                  std::numeric_limits<Real>::min(),
                                  "cylinder_axial_min>0",
                                  "Minimum axial coordinate");
```

will always fail for the default negative value now because that check says it must be positive.

Would you like me to change that range check function to something like `cylinder_axial_min<0` instead.

This would, of course, just say the value has to be negative instead of positive.

Then the only issue is that one test input problem [test/tests/meshdivisions/cylindrical_division.i](https://github.com/idaholab/moose/blob/next/test/tests/meshdivisions/cylindrical_division.i#L29) specifies:

```
[MeshDivisions]
  [cylindrical_div]
    type = CylindricalGridDivision
    . . .
    # Extent of the cylinder
    cylinder_axial_min = 0.5
    cylinder_axial_max = 2
    r_min = 0.5
    r_max = 4
  []
[]
```

And this positive `cylinder_axial_min` test input value then fails that `cylinder_axial_min<0` check with:

```
*** ERROR ***
Range check failed for parameter MeshDivisions/cylindrical_div/cylinder_axial_min
       Expression: cylinder_axial_min<0
       Value: 0.5
```

 - a) Should `cylinder_axial_min` really be a non-range-checked parameter that can be any value (-inf, +inf)?
 
 - b) Or should it be a range-checked negative value and [test/tests/meshdivisions/cylindrical_division.i](https://github.com/idaholab/moose/blob/next/test/tests/meshdivisions/cylindrical_division.i#L29) is bad?

Which of those above choices should be the change here? If (b) is the correct change, then with:

```
cylinder_axial_min = 0.5
```

in that test input, what do you suggest I change it to? Should this be the new value:

```
cylinder_axial_min = -0.5
```

Reply:
no range check is the way to go here.

and in the constructor we should check min < max that s it

Reply:
@GiudGiud -

Okay cool thanks, I'll make this update. And that `_min_z < _max_z` error check is already in the constructor.

Last question, I guess this `cylinder_axial_max` should also be changed to be a non-range-checked parameter?

```
  params.addRangeCheckedParam<Real>("cylinder_axial_max",
                                    std::numeric_limits<Real>::max(),
                                    "cylinder_axial_max>0",
                                    "Maximum axial coordinate");
```

And its value can be also be any real number positive or negative (as long as min < max), right?

Reply:
yes that seems good too.

Reply:
@GiudGiud -

My latest [commit](https://github.com/idaholab/moose/pull/26069/commits/968f8c6b6be1eec24a7b64d664d296370753b50f) just made these fixes in regard to those cylindrical axial parameters:

 - fixed the non-infinite size check to use `_max_z == std::numeric_limits<Real>::max()` instead of `min()`
 - updated `cylinder_axial_min` default value to `std::numeric_limits<Real>::lowest()` instead of `min()`
 - updated `cylinder_axial_min` to be a regular type parameter instead of a range-checked type parameter
 - updated `cylinder_axial_max` to be a regular type parameter instead of a range-checked type parameter
 
 Could you double-check that commit to make sure all looks good there?
 
- - - - -

@loganharbour -

I fixed the SHA in `versioner_hashes.yaml` too, since I rebased on devel which changed the WASP commit.

That should fix the `tests.versioner` failure, so this is good to again from my side if @GiudGiud approves.